### ISSUE TITLE
[BUG] Specific script not working after the 2.13.1 update

### ISSUE THREAD

**Issue by vyneer** [violentmonkey/violentmonkey#1572]
- [v] I have searched for existing issues that already reported this problem and found none

**Describe the bug**
[A userscript that I'm working on](https://github.com/vyneer/dgg-chat-gui-scripts/blob/main/dgg-utilities.user.js) stopped working after the 2.13.1 update. The issue appears when the page the script is supposed to work on is in an iframe.

**To Reproduce**
1. Install userscript
2. Go to https://www.destiny.gg/bigscreen
3. Script doesn't work and, additionally, is gone from the "Sub-frames only scripts" dropdown in the ViolentMonkey popup.

**Expected behavior**
The provided userscript is functional (bottom of the chat iframe having the movie clapper icon is a clear sign of the script at least loading).  
![image](https://user-images.githubusercontent.com/41237021/191040049-71dd2868-be7c-4f61-905b-0a033bf3661d.png)

**Actual behavior**
Script doesn't load.

<details>
  <summary>Relevant console log from the page</summary>

  ```
VM5166 injected.js:1 Uncaught (in promise) TypeError: Oe.post is not a function
    at bt (VM5166 injected.js:1:11120)
    at l (injected.js:1:8703)
    at ct (injected.js:1:8416)
    at dt (injected.js:1:9167)
    at qt (injected.js:1:14512)
```

![image](https://user-images.githubusercontent.com/41237021/191040658-f58d55d3-5d58-4ab2-a170-3bc687a9227f.png)

</details>

**What is the result in the [upcoming release](https://github.com/violentmonkey/violentmonkey/releases)?**
The bug persists.

**Environment:**
- OS: Arch Linux
- Browser: Google Chrome
- Browser Version: 105.0.5195.125
- Violentmonkey Version: 2.13.1

However, I had multiple other people report the bug to me that were using Windows, so it's not OS-specific (as far as I can tell).

**Additional context**
The issue is specific to Chrome (or, perhaps, all Chromium-based browsers, but I haven't personally checked), everything's perfect on Firefox. Just to help figure out the problem, it seems like the specific beta release that this issue appeared is [beta v2.13.0.15](https://github.com/violentmonkey/violentmonkey/releases/tag/v2.13.0.15), because beta v2.13.0.14 works fine.

It also seems like using ```// @inject-into content``` or adding an additional ```// @include``` pointing to the main page referenced earlier solves the issue, but breaks some other aspects of the script, so ideally i'd like to avoid that.

**Comment by tophf** [comment]
Fixed in 640ff5fe.

As a workaround until the next version is released you can add a `@match` for the main page and add `if (window==top) return`

**Comment by vyneer** [comment]
Thank you for the quick fix! :heart:
